kernel/include/lapic.h

```diff
@@ -0,0 +1,83 @@
+// https://habr.com/en/post/446312/ Has a nice graphical overview of APICs
+// https://wiki.osdev.org/APIC
+
+#pragma once
+#include <stdint.h>
+
+#define XVIAligned __attribute__((aligned(0x10)))
```

Comment:
Macros should be all capital letters. But this macro doesn't really save that much space and it pollutes the global namespace with a very specific macro. You should probably just use the attribute by itself.